app/models/host/base.rb

```diff
@@ -321,6 +323,26 @@ def import_missing_ids
     # Provide _id aliases for consistency with the _name methods
     alias_attribute :hardware_model_id, :model_id
 
+    def locked?
```

Comment:
if you would create expired at (e.g. this host valid only for one week) i think it would look nearly identical ?